examples/tpm2-integration-test/integration_test.go

```diff
@@ -0,0 +1,57 @@
+package tpm2_integration_test
+
+import (
+	"bytes"
+	"flag"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+var (
+	tpmPath = flag.String(
+		"tpm-path",
+		"",
+		"Path to TPM character device. Most Linux systems expose it under /dev/tpm0. "+
+			"Empty value (default) will disable all integration tests.")
+
+	binPath = flag.String(
+		"bin-path",
+		"./",
+		"The path to the directory where the binaries for the examples live.")
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *tpmPath == "" {
+		os.Exit(0)
+	} else {
+		os.Exit(m.Run())
+	}
+}
+
+func TestExtendPcr(t *testing.T) {
+	pcr1ValBefore := runTpm2Cmd(t, "readpcr", "-pcr", "1")
+	runTpm2Cmd(t, "extendpcr", "-pcr", "1", "-data", "FF")
+	pcr1ValAfter := runTpm2Cmd(t, "readpcr", "-pcr", "1")
+	if pcr1ValBefore == pcr1ValAfter {
+		t.Fail()
```

Comment:
Use t.Fatalf to log the values you saw and explaining why this is a failure

Reply:
I could add something like:

```
		t.Fatalf(
			"Expected PCR 1 to have changed value after extending it. " +
			"Value before: %q Value after: %q", pcr1ValBefore, pcr1ValAfter)
```

But this simply re-states the test:

```
	pcr1ValBefore := runTpm2Cmd(t, "readpcr", "-pcr", "1")
	runTpm2Cmd(t, "extendpcr", "-pcr", "1", "-data", "FF")
	pcr1ValAfter := runTpm2Cmd(t, "readpcr", "-pcr", "1")
	if pcr1ValBefore == pcr1ValAfter {
```

If the test fails, it will be because extend had a bug in it (command succeeds but doesn't extend) or one of the commands return a non zero code. In either case the proposed extra debug message doesn't give any useful information (the particular value returned by ReadPCR is effectively random...I should add a separate test for ReadPCR though...I'll do that).

Reply:
You're right in that it doesn't five any extra information.
But it should produce a meaningful error message without forcing caller to read and parse the test code.